tests/epyccel/test_epyccel_sets.py

```diff
@@ -520,3 +520,33 @@ def union_int():
     pyccel_result = epyccel_func()
     python_result = union_int()
     assert python_result == pyccel_result
+
+def test_set_iter(language):
```

Comment:
This doesn't seem to work for  `C` and `Fortran`. There is a pytest param on line 488 that seems to restrict language to Python. I think handling iteration for all languages is within the scope of this PR.

Reply:
The param on line 488 is a parametrize decorator. It only applies to the function that it is decorating

Reply:
Here is proof that this works for C and Fortran:
![image](https://github.com/user-attachments/assets/b5792820-cdb6-4c9c-a7b0-0b82e8890c02)

Reply:
I apologize profusely for the confusion.
I tested this and it did not work:
```python
if __name__ == '__main__':
    a = {1,2,3}
    ss = 0
    for x in a:
        ss += x
    print(ss)
```
that's why i thought that the tests did not run.
but then, why the following works but not the former:
```python
def f():
    a = {1,2,3}
    ss = 0
    for x in a:
        ss += x
    return ss

if __name__ == '__main__':
    print(f())
```

Reply:
What error are you seeing? This code seems to work fine for me

Reply:
So sorry, I had to delete __pyccel__, something was cached and was showing this error:
```
ERROR at code generation stage
pyccel:
 |error [codegen]: boo.py| Don't know how to index int type
Pyccel has encountered syntax that has not been implemented yet. Please create an issue at https://github.com/pyccel/pyccel/issues and provide a small example of your problem. Do not forget to specify your target language.
```

Reply:
That is the error on the devel branch. Very strange if you're seeing it here. I would guess that you maybe didn't have pyccel installed in editable mode but if that's the case then I've no idea how you've been developing ^^

Reply:
I do have it installed in editable mode. The issue had to do with the `__pyccel__` folder. It does not seem related but deleting it fixed the issue.